packages/core/src/split-layout/SplitLayout.tsx

```diff
@@ -0,0 +1,70 @@
+import { ElementType, forwardRef, ReactElement, ReactNode } from "react";
+import { FlexLayout, FlexLayoutProps } from "../flex-layout";
+import {
+  PolymorphicComponentPropWithRef,
+  PolymorphicRef,
+  useIsViewportLargerThanBreakpoint,
+} from "../utils";
+import { Breakpoints } from "../breakpoints";
+
+export type SplitLayoutProps<T extends ElementType> =
+  PolymorphicComponentPropWithRef<
+    T,
+    {
+      /**
+       * Defines the default behavior for how flex items are laid out along the cross axis on the current line.
+       */
+      align?: FlexLayoutProps<ElementType>["align"];
+      /**
+       * Establishes the main-axis, defining the direction children are placed. Default is "row".
+       */
+      direction?: FlexLayoutProps<ElementType>["direction"];
+      /**
+       * End component to be rendered.
+       */
+      endItem?: ReactNode;
+      /**
+       * Controls the space between left and right items.
+       */
+      gap?: FlexLayoutProps<ElementType>["gap"];
+      /**
+       * Start component to be rendered.
+       */
+      startItem?: ReactNode;
+      /**
+       * Breakpoint at which the horizontal split wraps.
+       */
+      wrapAtBreakpoint?: keyof Breakpoints;
+    }
+  >;
+type SplitLayoutComponent = <T extends ElementType = "div">(
+  props: SplitLayoutProps<T>
+) => ReactElement | null;
+
+export const SplitLayout: SplitLayoutComponent = forwardRef(
+  <T extends ElementType = "div">(
+    {
+      direction,
+      endItem,
+      startItem,
+      wrapAtBreakpoint = "sm",
```

Comment:
I'd have thought by default we would leave `wrapAtBreakpoint` as undefined. What do you think?

Reply:
I added `sm` because I removed wrap all together to simplify the props, would you say we would want to have `wrap` and `wrapAtBreakpoint` ? to me it felt like as a user i could feel like i need both on, i guess `wrapAtBreakpoint` is strictly not really wrapping but changing direction, also this approach doesn't let you not wrap... 
i guess options are: 
1) wrap (boolean) and wrapAtBreakpoint(breakpoint, default undefined)
2) wrapAtBreakpoint(breakpoint, default undefined means no wrap)
3) wrapAtBreakpoint(breakpoint,  add a `default` option that does wrap, and undefined means no wrap)

Reply:
I guess another option is to have neither prop as the user can user a responsive prop with direction?